plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/_impl.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+
+import nevergrad as ng
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.override_parser.types import (
+    ChoiceSweep,
+    IntervalSweep,
+    Override,
+    Transformer,
+)
+from hydra.core.plugins import Plugins
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.sweeper import Sweeper
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, ListConfig, OmegaConf
+
+from .config import OptimConf, ScalarConfigSpec
+
+# pylint: disable=logging-fstring-interpolation,no-self-used
+log = logging.getLogger(__name__)
+
+
+def verify_scalar_bounds(scalar: ng.p.Scalar) -> None:
+    if scalar.integer:
+        a, b = scalar.bounds
+        if a is not None and b is not None and b - a <= 6:
+            raise ValueError(
+                "For integers with 6 or fewer values, use a choice instead"
+            )
+
+
+def create_nevergrad_param_from_config(
+    config: Union[ListConfig, DictConfig, List[Any], Dict[str, Any], str]
```

Comment:
The function implementation is not doing anything if the type is Dict or List.
is this intentional?
can you just not call it in this case?

Reply:
I'm not sure what you mean by not doing anything, it actually wraps the the values in a ng param right?

Reply:
Ah, I missed it. the function signature is a bit cumbersome.
this would probably work:

```
def create_nevergrad_param_from_config(config: Union[MutableSequence[Any], MutableMapping[str, Any]]
```

Also, you declare that the input can also be a string but I am not seeing any handling.

Reply:
thx, `str` shouldn't be there, removed.

Thx for the typing tips! much cleaner now!